crates/adapters/src/controller/mod.rs

```diff
@@ -228,6 +230,26 @@ impl Controller {
         Ok(())
     }
 
+    /// Check whether the pipeline has processed all input data to completion.
+    ///
+    /// This method should only be used with pipelines whose input endpoints
```

Comment:
what happens if it's not?
won't it just return 'false'?

Reply:
you're right